### PR TITLE
Implement BatchStatus using FIFO

### LINKED ISSUE #1986: Implement BatchStatus using FIFO

### PR DESCRIPTION
**Pull request by cmcantalupo**
- Relates to #1962
- Fixes #1986

### CODE REVIEW

File: service/src/BatchStatus.hpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            int m_read_fd;
+            int m_write_fd;
+    };
+
+    class BatchStatusServer : public BatchStatusImp
+    {
+        public:
+            /**
+             * The constructor which is called by the server.
+             */
```
Comment (cmcantalupo) [inline service/src/BatchStatus.hpp]:
```suggestion
            ///
            /// The constructor which is called by the server.
            ///
```

File: service/src/BatchStatus.hpp [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            std::string m_read_fifo_path;
+            std::string m_write_fifo_path;
+    };
+
+    class BatchStatusClient : public BatchStatusImp
+    {
+        public:
+            /**
+             * The constructor which is called by the client.
+             */
```
Comment (cmcantalupo) [inline service/src/BatchStatus.hpp]:
```suggestion
            ///
            /// The constructor which is called by the client.
            ///
```

File: service/src/BatchStatus.cpp [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    /***********************************
+     * Members of class BatchStatusImp *
+     ***********************************/
+
+    BatchStatusImp::BatchStatusImp(int m_read_fd, int m_write_fd)
+    : BatchStatus{},
+      m_read_fd{m_read_fd},
+      m_write_fd{m_write_fd}
```
Comment (cmcantalupo) [inline service/src/BatchStatus.cpp]:
```suggestion
    : BatchStatus{}
    , m_read_fd{m_read_fd}
    , m_write_fd{m_write_fd}
```

File: service/src/BatchStatus.cpp [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+    /**************************************
+     * Members of class BatchStatusServer *
+     **************************************/
+
+    // The constructor which is called by the server.
+    BatchStatusServer::BatchStatusServer(int client_pid, const std::string &server_key)
+    : BatchStatusImp(-1, -1),
+      m_read_fifo_path(std::string(M_FIFO_PREFIX) + "-in-" + server_key),
+      m_write_fifo_path(std::string(M_FIFO_PREFIX) + "-out-" + server_key)
```
Comment (cmcantalupo) [inline service/src/BatchStatus.cpp]:
```suggestion
    : BatchStatusImp(-1, -1)
    , m_read_fifo_path(std::string(M_FIFO_PREFIX) + "-in-" + server_key)
    , m_write_fifo_path(std::string(M_FIFO_PREFIX) + "-out-" + server_key)
```

File: service/src/BatchStatus.cpp [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+    /**************************************
+     * Members of class BatchStatusClient *
+     **************************************/
+
+    // The constructor which is called by the client.
+    BatchStatusClient::BatchStatusClient(const std::string &server_key)
+    : BatchStatusImp(-1, -1),
+      m_read_fifo_path(std::string(M_FIFO_PREFIX) + "-out-" + server_key),
+      m_write_fifo_path(std::string(M_FIFO_PREFIX) + "-in-" + server_key)
```
Comment (cmcantalupo) [inline service/src/BatchStatus.cpp]:
```suggestion
    : BatchStatusImp(-1, -1)
    , m_read_fifo_path(std::string(M_FIFO_PREFIX) + "-out-" + server_key)
    , m_write_fifo_path(std::string(M_FIFO_PREFIX) + "-in-" + server_key)
```

File: service/test/BatchStatusTest.cpp [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+};
+
+void BatchStatusTest::SetUp(void)
+{
+    m_server_prefix = BatchStatus::M_FIFO_PREFIX;
+    m_server_key = "test-key";
+
+    /**********************************************************************
+     * Explicitly force the fifo to be removed if it is already existing. *
+     **********************************************************************/
```
Comment (Krebrov001) [inline service/test/BatchStatusTest.cpp]: Change to regular comment.

**Review by Krebrov001 — DISMISSED** [review]
All changes have been implmeneted.

**Review by Krebrov001 — APPROVED** [review]
# Approve the changes.

**Comment by Krebrov001** [comment]
This fixes #1976

**Pull request merged**